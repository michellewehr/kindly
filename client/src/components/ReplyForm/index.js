import { useState } from "react";
import { useMutation } from "@apollo/client";
import { ADD_REPLY } from "../../utils/mutations";

export default function ReplyForm({commentId}) {
  const [replyBody, setReplyBody] = useState("");
  const [characterCount, setCharacterCount] = useState(0);
  const [addReply] = useMutation(ADD_REPLY);

  const handleChange = (event) => {
    if (event.target.value.length <= 280) {
      setReplyBody(event.target.value);
      setCharacterCount(event.target.value.length);
    }
  };

  const handleFormSubmit = async (event) => {
    event.preventDefault();
    console.log(replyBody, 'reply body')
    console.log(commentId);
    try {
      // ! add reply to database
      await addReply({
        variables: { replyBody, commentId}
      });

      // clear form value
      setReplyBody("");
      setCharacterCount(0);
    } catch (e) {
      console.error(e);
    }
  };

  return (
    <div>
      <p
      // className={`m-0 ${characterCount === 280 || error ? "text-error" : ""}`}
      >
        Character Count: {characterCount}/280
        {/* {error && <span className="ml-2">Something went wrong...</span>} */}
      </p>
      <form
        className=""
        onSubmit={handleFormSubmit}
      >
        <textarea
          placeholder="reply to this comment..."
          value={replyBody}
          className="form-input col-12 col-md-9"
          onChange={handleChange}
        ></textarea>

        <button className="btn col-12 col-md-3" type="submit">
          Submit
        </button>
      </form>
    </div>
  );
}
