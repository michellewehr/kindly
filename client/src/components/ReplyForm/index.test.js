import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ReplyForm from './index';

afterEach(cleanup);

describe('ReplyForm component renders', () => {

   it('renders', () => {
      render(<ReplyForm />);
   });

   it('Matches snapshot DOM node structure', () => {
      const { asFragment } = render(<ReplyForm />);
      expect(asFragment()).toMatchSnapshot();
   });
});