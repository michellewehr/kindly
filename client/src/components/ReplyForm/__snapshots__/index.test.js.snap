// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReplyForm component renders Matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    <p>
      Character Count: 0/280
    </p>
    <form
      class=""
    >
      <textarea
        class="form-input col-12 col-md-9"
        placeholder="reply to this comment..."
      />
      <button
        class="btn col-12 col-md-3"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
